platform/net/ubuntu_net_manager.go

```diff
@@ -164,20 +166,12 @@ func (net UbuntuNetManager) SetupNetworking(networks boshsettings.Networks, mbus
 		return bosherr.WrapError(err, "Validating dns configuration")
 	}
 
-	go net.addressBroadcaster.BroadcastMACAddresses(append(staticAddressesWithoutVirtual, dynamicAddresses...))
-	err = net.setupFirewall(mbus)
+	err = SetupNatsFirewall(mbus)
 	if err != nil {
-		return bosherr.WrapError(err, "Setting up Nats Firewall")
-	}
-	return nil
-}
-
-func (net UbuntuNetManager) setupFirewall(mbus string) error {
-	if mbus == "" {
-		net.logger.Info("NetworkSetup", "Skipping adding Firewall for outgoing nats. Mbus url is empty")
-		return nil
+		return bosherr.WrapError(err, "Setting up nats firewall")
 	}
-	net.logger.Info("NetworkSetup", "Adding Firewall not implemented on")
+	net.logger.Info(UbuntuNetManagerLogTag, "Successfully set up outgoing nats api firewall")
+	go net.addressBroadcaster.BroadcastMACAddresses(append(staticAddressesWithoutVirtual, dynamicAddresses...))
```

Comment:
Why we moved the IPs broadcast to the end? It was before the `setupFirewall` call https://github.com/cloudfoundry/bosh-agent/pull/264/commits/7d56f583f8806b1b45d179c834e03e9deba94b0c#diff-010e8a578b784cca80730f6e56600727eede1f1fafdd186f718872bff04841e6L167

Reply:
same as above. I just put in skelethon code when the windows counterpart was written.